### ISSUE TITLE
sql: performance difference between forward (ASC) and reverse scans (DESC)

### ISSUE THREAD

**Issue by jseldess** [cockroachdb/docs#1195]
Jesse Seldess (jseldess) commented:

Call out in `SELECT` as well as troubleshooting/SQL optmization docs.

See https://github.com/cockroachdb/cockroach/issues/13935#issuecomment-286895436 for details.

**Comment by tschottdorf** [comment]
Also depends on the encoding - if keys are encoded in descending order, then a reverse scan is the better one (because on the kv level it turns once again into a forward one). I.e. going against the index direction is what's slow.

**Comment by a-robinson** [comment]
Yeah, just to second @tschottdorf, If the index is sorted in DESC order, then DESC SELECTs will be fast. It's just when the SELECT order doesn't match the index order that things are slower.

**Comment by jseldess** [comment]
Thanks for the clarifications, @a-robinson and @tschottdorf.

**Comment by petermattis** [comment]
I'm surprised reverse iteration is so much slower. Internally, RocksDB caches the decoded block the iterator is currently pointed at. So accessing the next or previous key should be about the same speed. But now I'm remember that `MVCCIterate` is a bit slower for reverse iteration though I can't recall why.

**Comment by jseldess** [comment]
We have closed this issue because it is more than 3 years
old. If this issue is still relevant, please add a comment and reopen
the issue. Thank you for your contribution to CockroachDB docs!
